src/test/java/com/monumental/services/MonumentServiceTests.java

```diff
@@ -0,0 +1,175 @@
+package com.monumental.services;
+
+import com.monumental.models.Monument;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MonumentServiceTests {
+
+    @Mock
+    private SessionFactory sessionFactory;
+
+    @Mock
+    private SessionFactoryService sessionFactoryService;
+
+    @Mock
+    private Session session;
+
+    @Mock
+    private Transaction transaction;
+
+    @Mock
+    private Query getAllQuery;
+
+    private Monument monument1;
+    private Monument monument2;
+    private Monument monument3;
+
+    private List<Monument> monuments;
+    private List<Integer> ids;
+
+    private MonumentService monumentService;
+
+    /**
+     * Setup method that is run before tests are executed
+     * Sets up the various variables and mocks used during the tests
+     */
+    @Before
+    public void setUp() {
+        // Setup MonumentService for tests
+        this.monumentService = new MonumentService(this.sessionFactoryService);
+
+        // Setup Monument objects for tests
+        this.monument1 = new Monument();
+        this.monument1.setTitle("Monument");
+
+        this.monument2 = new Monument();
+        this.monument2.setTitle("Monument");
+
+        this.monument3 = new Monument();
+        this.monument3.setTitle("Monument");
+
+        this.monuments = new ArrayList<>();
+        this.monuments.add(this.monument1);
+        this.monuments.add(this.monument2);
+        this.monuments.add(this.monument3);
+
+        this.ids = new ArrayList<>();
+        this.ids.add(1);
+        this.ids.add(2);
+        this.ids.add(3);
+
+        // Setup SessionFactoryService mock
+        when(this.sessionFactoryService.getFactory()).thenReturn(this.sessionFactory);
+        when(this.sessionFactoryService.getFactory().openSession()).thenReturn(this.session);
+
+        // Setup Session mock
+        when(this.session.beginTransaction()).thenReturn(this.transaction);
+
+        when(this.session.save(this.monument1)).thenReturn(1);
+        when(this.session.save(this.monument2)).thenReturn(2);
+        when(this.session.save(this.monument3)).thenReturn(3);
```

Comment:
I think this is a kind of unfortunate drawback of mockito.

Another way of doing this is with an in-memory database, which lets you actually run SQL and see the results without affecting your local DB https://www.baeldung.com/spring-jpa-test-in-memory-database